### ISSUE TITLE
Illegal Instructions when trying to load kohya_ss on Ubuntu

### ISSUE THREAD

**Issue by samuelkurt** [bmaltais/kohya_ss#2439]
![Screenshot from 2024-05-03 21-33-42](https://github.com/bmaltais/kohya_ss/assets/104158296/7bfd35fb-e92c-4669-adf2-1d10d80ebb0e)
I've installed all required Software and selected the right installation Type. I don't know what causes that Error, but I think i did everything correctly.

**Comment by bmaltais** [comment]
I have never seen this before... This can be tricky to troubleshoot... I only test the linuix part on WSL2... what version of Ubuntu do you use? I use 22.04 in WSL2. This look to be something that comes from gui.sh on line 120...

![image](https://github.com/bmaltais/kohya_ss/assets/7474674/237123b0-5198-4685-affa-7c41fe557433)

I don't understand why your install of Ubuntu is having this issue...

**Comment by samuelkurt** [comment]
I use Ubuntu 22.04, Dualboot with Windows.

**Comment by bmaltais** [comment]
Unless I can reproduce the issue I can't really fix it... But this appear to me to be specific to your installation or many others would have complained of the same issue. But what is causing this in your case, I have no idea.

**Comment by samuelkurt** [comment]
I don't know either. Is there a Linux distro you would recommend? Except Ubuntu in WSL2 mode.

**Comment by bmaltais** [comment]
Ubuntu 22.04 is what I would have recommended…

**Comment by samuelkurt** [comment]
I am on Ubuntu 22.04, guess there’s nothing I can do about it.

**Comment by samuelkurt** [comment]
What are all the requierments I need to install, maybe i've just forgot on.

**Comment by bmaltais** [comment]
It should be those as described in the README... I have never tested running on a native Linux Ubuntu install... only on WSL2 Ubuntu 22.04...

**Comment by samuelkurt** [comment]
Is there a difference between WSL and WSL2 because I tried it on WSL 22.04 and i still got the same Error

**Comment by bmaltais** [comment]
It has to be WSL2 ( the recent version on win11)

**Comment by samuelkurt** [comment]
Downgrading to kohya_ss  v23.0.15 fixed it.
![Screenshot from 2024-05-12 18-38-57](https://github.com/bmaltais/kohya_ss/assets/104158296/3d98bf3f-130c-462e-8af0-533b9a91aece)

**Comment by bmaltais** [comment]
This is a good find. There might have been an update to the gui.sh script causing this. With this I do I can do a compare between the current version and this one to see what might be causing this.

**Comment by FunRoss7** [comment]
I encountered the same problem on Ubuntu 22.04.1, and am now trying the rollback.

**Comment by bmaltais** [comment]
Setup.sh is actually the same berween both release... so the issue is nor in that script... but why is is failing in the recent release... hard to tell since I can't reproduce it.

**Comment by bmaltais** [comment]
THe issue is with the validation.py script that validate everything is fine before running the GUI... my theory.

**Comment by samuelkurt** [comment]
I just discovered that kohya_ss 24.0.4 boots up without any issues, while kohya_ss 24.0.5 throws the exact same Error Message. So the Bug must have happened  between those two versions.
![Screenshot from 2024-05-14 20-01-51](https://github.com/bmaltais/kohya_ss/assets/104158296/349aef64-e63e-4c84-8e9d-2a7128df160a)
![Screenshot from 2024-05-14 20-03-55](https://github.com/bmaltais/kohya_ss/assets/104158296/e6994025-43f0-4b7f-ae0d-e5d9c107acc3)

**Comment by bmaltais** [comment]
The significant difference between them is the switch to utf-8 vs utf8 file encoding… not sure why this would be an issue…

**Comment by bmaltais** [comment]
I created a test branch with utf8 instead of utf-8... can you give it a try:

```
git fetch origin
git checkout utf8
git pull
```

**Comment by samuelkurt** [comment]
That still hasn't fixed it.

**Comment by bmaltais** [comment]
Ok, so at least the obvious changes are not the source… I will look for the next best culprit…

**Comment by b-fission** [comment]
Found the source of the issue.

https://github.com/bmaltais/kohya_ss/blob/92a01a3890d86f10973fe0f18a79b1d50e5d3b83/kohya_gui/class_tensorboard.py#L9

I tested on a VM with AVX disabled. Under those conditions, importing tensorflow indeed causes python to crash with the same `Illegal instruction` error being reported here.

Perhaps the `import tensorflow` check can be done through a subprocess to avoid a full crash.
